src/elf.rs

```diff
@@ -570,9 +577,22 @@ impl<'a> Elf<'a> {
                 }
             }
             DefinedDecl::Section(_) => {
-                for (_symbol, _symbol_dst_offset) in def.symbols {
-                    // FIXME: implement it
-                    unimplemented!("elf: custom symbols referencing sections");
+                for (symbol, symbol_dst_offset) in def.symbols {
+                    // TODO: can we move the string instead of cloning?
+                    let (idx, offset) = self.new_string(symbol.clone());
+                    debug!(
+                        "idx: {:?} @ {:#x} - new strtab offset: {:#x}",
+                        idx, offset, self.sizeof_strtab
+                    );
+
+                    // TODO: per-symbol type?
```

Comment:
could you clarify what this TODO means?

Reply:
I'm passing `SymbolType::None` immediately below. The TODO is supposed to be a reminder to think about whether we want to be able to define that on a per-symbol basis.

I.e., say you're producing a section that is a function and its associated constants. You give the function a symbol of type Function and the constants a symbol each of type Data, just because you can. Currently, you can only pass in "there should be a symbol located here", but not whether it's pointing to a function or to some data or something.

It may not belong there, "TODO" is how I leave notes to myself when I have questions about design or see potential future features (that i don't want to try implementing right now), whether I'm working on my code or someone else's.